xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocuments.xml

```diff
@@ -343,76 +369,60 @@ XWiki.index.trash.documents.confirmDelete = function(event, table, rowIdx) {
- */
-XWiki.index.trash.documents.confirmDeleteAll = function(event) {
-  event.stop();
-  if (this.hasClassName('disabled')) {
-    // Do nothing if the button was already clicked and it's waiting for a response from the server.
-    return;
-  } else {
-    new XWiki.widgets.ConfirmedAjaxRequest(
-      /* Ajax request URL */
-      XWiki.currentDocument.getURL('delete'),
-      /* Ajax request parameters */
-      {
-        parameters : {
-          'confirm': 1,
-          'emptybin': true,
-          'form_token': "$!{services.csrf.getToken()}",
-          'ajax': 1
-        },
-        onCreate : function() {
-          // Disable the button, to avoid a cascade of clicks from impatient users
-          this.addClassName('disabled');
-        }.bind(this),
-        onSuccess : function() {
-          // Update the livetable
-          trashDocumentsLivetable.refresh();
-        }.bindAsEventListener(this)
-      },
-      /* Interaction parameters */
-      {
-         confirmationText: "$services.localization.render('core.recyclebin.completelyDeleteConfirm')",
-         progressMessageText : "$services.localization.render('platform.index.trashDocumentsDeleteInProgress')",
-         successMessageText : "$services.localization.render('platform.index.trashDocumentsDeleteDone')",
-         failureMessageText : "$services.localization.render('platform.index.trashDocumentsDeleteFailed')"
-      }
-      );
-  }
-}
-/*
- * The livetable is hidden when javascript is disabled, so that an alternative &lt;noscript&gt; table is displayed.
- * Re-display it when the document loads and Javascript is available.
- */
-document.observe('xwiki:dom:loaded', function() {
-  if ($('documentsTrash')) {
-    $('documentsTrash').up('.hidden').removeClassName('hidden');
-  }
+require(['jquery', 'xwiki-meta', 'xwiki-events-bridge'], function($, xm) {
+  //
+  // Delete All
+  //
+  var deleteDocumentConfirmModal = $('#deleteDocumentConfirmModal');
+  deleteDocumentConfirmModal.on('show.bs.modal').find('.btn-danger').click(function() {
+    var notification = new XWiki.widgets.Notification(
+      $jsontool.serialize($services.localization.render('platform.index.trashDocumentsDeleteInProgress')),
+      'inprogress'
+    );
+    $.post(XWiki.currentDocument.getURL('delete'), {
+        'confirm': 1,
+        'emptybin': true,
+        'form_token': xm.form_token,
+        'ajax': 1
+    }).done(function() {
+      notification.replace(new XWiki.widgets.Notification(
+        $jsontool.serialize($services.localization.render('platform.index.trashDocumentsDeleteDone')),
+        'done'
+      ));
+      trashDocumentsLivetable.refresh();
+    }).fail(function() {
+      notification.replace(new XWiki.widgets.Notification(
+        $jsontool.serialize($services.localization.render('platform.index.trashDocumentsDeleteFailed')),
+        'error'
+      ));
+    });
+  });
+  $(document).on('xwiki:livetable:documentsTrash:displayComplete', function(event, data) {
```

Comment:
You don't use the ``event`` and ``data`` parameters below so you can move the code in a function that is called:
* initially, if ``trashDocumentsLivetable`` is defined and if ``$('#documentsTrash td.xwiki-livetable-display-header-filter:last-child')`` is not empty (that is if the live table was loaded before this code)
* on ``xwiki:livetable:documentsTrash:displayComplete`` (if the live table is not yet loaded and for the future reloads of the live table)

Reply:
OK so basically I extract the function, I attach it as a handler for the event, and I also call it at the end of the JS to be evaluated once in case the livetable is already loaded.